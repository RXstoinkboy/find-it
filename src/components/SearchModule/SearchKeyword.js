import React, {Component} from 'react';
import styles from './styles/SearchInput.module.scss'
import PropTypes from 'prop-types'

import ResultsMod from './ResultsMod'

export class SearchKeyword extends Component {
    constructor(props){
        super(props);
        this.state = {
            focus: false
        }
    }

    toggleFocus = e => {
        this.setState({
            focus: !this.state.focus
        })
    }

    render(){
        return (
            <div className={styles.wrapper}>
                <input 
                    type="search" 
                    placeholder='Search by name, category etc...' 
                    onChange={this.props.handleSearch}
                    onFocus={this.toggleFocus}
                    onBlur={this.toggleFocus}
                    value={this.props.keywordToSearch}
                    className={styles.searchInput} />
                <ResultsMod {...this.props} focus={this.state.focus}/>
            </div>
        );
    }
};

export default SearchKeyword;

SearchKeyword.propTypes = {
    handleSearch: PropTypes.func.isRequired,
    keywordToSearch: PropTypes.string.isRequired
}