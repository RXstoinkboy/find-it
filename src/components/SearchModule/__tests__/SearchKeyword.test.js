import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {SearchKeyword} from '../SearchKeyword'

it(`SearchKeyword renderd correctly`, ()=> {
    const props = {
        handleSearch: jest.fn(),
        keywordToSearch: 'wiel'
    }
    const searchKeyword = shallow(
            <SearchKeyword 
                {...props}
            />
        )
    expect(toJson(searchKeyword)).toMatchSnapshot()
})
