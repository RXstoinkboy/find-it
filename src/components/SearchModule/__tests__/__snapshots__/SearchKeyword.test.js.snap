// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchKeyword renderd correctly 1`] = `
<div
  className="wrapper"
>
  <input
    className="searchInput"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    placeholder="Search by name, category etc..."
    type="search"
    value="wiel"
  />
  <ResultsMod
    focus={false}
    handleSearch={[MockFunction]}
    keywordToSearch="wiel"
  />
</div>
`;
